source/SpinalHDL/Data types/bits.rst

```diff
@@ -41,7 +42,15 @@ The syntax to declare a bit vector is as follows: (everything between [] is opti
 .. code-block:: scala
 
    // Declaration
-   val myBits  = Bits()     // the size is inferred     
+   val myBits  = Bits()     // the size is inferred at first assignment
+   // ....
+   myBits := B("001010")    // Bits(6 bits) would be inferred
+   when(condxMaybe) {
+     myBits := B("110000")
+   } elsewhen(condxMaybeTwo) {
+     //myBits := B("00110000")	// Would Error: WIDTH MISMATCH (8 bits <- 6 bits)
```

Comment:
```suggestion
     //myBits := B("0011")	// Would Error: we can't assign a 4 bit Bits
     to a 6 bit Bits w/o resizing
```
We should use your original example with 4 bits as it does not change which size is inferred - easier to understand IMO